test/unit/spa.test.js

```diff
@@ -138,6 +138,22 @@ describe('spa', () => {
     consola.log.mockClear()
   })
 
+  test('render:route hook does not corrupt the cache', async () => {
+    const window1 = await nuxt.server.renderAndGetWindow(url('/'))
+    const html1 = window1.document.body.innerHTML
+    expect(html1).toContain('extra html from render:route hook')
+    expect(html1.match(/render:route/g).length).toBe(1)
+
+    window1.close()
+
+    const window2 = await nuxt.server.renderAndGetWindow(url('/'))
+    const html2 = window2.document.body.innerHTML
+    expect(html2).toContain('extra html from render:route hook')
+    expect(html2.match(/render:route/g).length).toBe(1)
```

Comment:
How does this test ensure modification? Shouldn't we ensure html1 and html2 to not be the same?

Extra point: jest has [toMatch](https://jest-bot.github.io/jest/docs/expect.html#tomatchregexporstring) helper.

Reply:
no, actually we want them to be different. without the patch, they are different, but the extra test is added multiple times.

on first request, we calculate the html and put it into the cache.
the hook modifies the data in the cache.

on second request, we get the html from the cache (with hook's modification) and hook runs again and adds the data another time.

What I'm trying to achieve here is the following:

- make sure the data is cached, but then modify it per request (per user etc). the modified response should not be put into the cache.

Reply:
The reason I didn't use toMatch is that I want to ensure there is a single match. Without the match, there is a match, but twice.

Let me give another example:

- First request, html is calculated and cached. Let's say html = "first request".
- Let's say we have a hook that adds: 'hook was here". Now the html returned (and html in cache - due to mutable js objects is) " first request hook was here"
- On second request, we get the html from the cache, append "hook was here", the result becomes "" first request hook was here hook was here"

And it keeps appending the same string on subsequent requests.

Reply:
Ok. I was confused by `new Date()`